setup.py

```diff
@@ -0,0 +1,210 @@
+    ]
+    return constrained_requirements
+
+
+def is_requirement(line):
+    """
+    Return True if the requirement line is a package requirement.
+
+    Returns:
+        bool: True if the line is not blank, a comment,
+        a URL, or an included file
+    """
+    return (
+        line and line.strip() and not line.startswith(("-r", "#", "-e", "git+", "-c"))
+    )
+
+
+def package_data(pkg, roots, sub_roots):
+    """
+    Declare package_data based on `roots`.
+
+    All of the files under each of the `roots` will be declared as package
+    data for package `pkg`.
+
+    """
+    data = []
+    for root in roots:
+        for sub_root in sub_roots:
+            for dirname, _, files in os.walk(os.path.join(pkg, root, sub_root)):
+                for fname in files:
+                    data.append(os.path.relpath(os.path.join(dirname, fname), pkg))
+
+    return {pkg: data}
+
+
+VERSION = get_version("xblocks_contrib", "__init__.py")
+
+if sys.argv[-1] == "tag":
+    print("Tagging the version on github:")
+    os.system("git tag -a %s -m 'version %s'" % (VERSION, VERSION))
+    os.system("git push --tags")
+    sys.exit()
+
+README = open(
+    os.path.join(os.path.dirname(__file__), "README.rst"), encoding="utf8"
+).read()
+CHANGELOG = open(
+    os.path.join(os.path.dirname(__file__), "CHANGELOG.rst"), encoding="utf8"
+).read()
+
+setup(
+    name="xblocks-contrib",
+    version=VERSION,
+    description="""core xblocks""",
+    long_description=README + "\n\n" + CHANGELOG,
+    author="Open edX Project",
+    author_email="oscm@openedx.org",
+    url="https://github.com/openedx/xblocks-contrib",
+    packages=find_packages(
+        include=["xblocks_contrib", "xblocks_contrib.*"],
+        exclude=["*tests"],
+    ),
+    include_package_data=True,
+    install_requires=load_requirements("requirements/base.in"),
+    python_requires=">=3.11",
+    license="AGPL 3.0",
+    zip_safe=False,
+    keywords="Python edx",
+    classifiers=[
+        "Development Status :: 3 - Alpha",
+        "Intended Audience :: Developers",
+        "License :: OSI Approved :: GNU Affero General Public License v3 or later (AGPLv3+)",
+        "Natural Language :: English",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.11",
+        "Programming Language :: Python :: 3.12",
+    ],
+    entry_points={
+        "xblock.v1": [
+            # _xblock suffix is added for testing only.
+            "annotatable_xblock = xblocks_contrib:AnnotatableBlock",
+            "discussion_xblock = xblocks_contrib:DiscussionXBlock",
+            "html_xblock = xblocks_contrib:HtmlBlock",
+            "lti_xblock = xblocks_contrib:LTIBlock",
+            "poll_question_xblock = xblocks_contrib:PollBlock",
+            "problem_xblock = xblocks_contrib:ProblemBlock",
+            "video_xblock = xblocks_contrib:VideoBlock",
+            "word_cloud_xblock = xblocks_contrib:WordCloudBlock",
```

Comment:
I like that you included these entrypoints and used a suffix to differentiate them. Instead of `{name}_xblock`, could you do `_{name}_extracted`? That'd make it clear when someone has installed an extracted block rather than the original block. The underscore would help indicate that it's an unstable entrypoint, which is important, since we don't want sites running Sumac to mistake these for stable block tags and write content that depends on them.